EIPS/eip-721.md

```diff
@@ -0,0 +1,404 @@
+    ///  (`to` == 0). Exception: during contract creation, any number of deeds
+    ///  may be created and assigned without emitting Transfer. At the time of
+    ///  any transfer, the approved address for that deed (if any) is reset to none.
+    event Transfer(address indexed _from, address indexed _to, uint256 _deedId);
+
+    /// @dev This emits when the approved address for a single is changed or
+    ///  reaffirmed. The zero address indicates there is no approved address.
+    ///  When a Transfer event emits, this also indicates that the approved
+    ///  address for that deed (if any) is reset to none.
+    event Approval(address indexed _owner, address indexed _approved, uint256 _deedId);
+
+    /// @dev This emits when an operator is enabled or disable for an owner.
+    ///  The operator may manage all deeds of the owner.
+    event ApprovalForAll(address indexed _owner, address indexed _operator, bool _approved);
+
+    /// @notice Count all deeds assigned to an owner
+    /// @dev Deeds assigned to zero address are considered invalid, and this
+    ///  function throws for queries about the zero address.
+    /// @param _owner An address for whom to query the balance
+    /// @return The number of deeds owned by `_owner`, possibly zero
+    function balanceOf(address _owner) external view returns (uint256 _balance);
+
+    /// @notice Find the owner of a deed
+    /// @param _deedId The identifier for a deed we are inspecting
+    /// @dev Deeds assigned to zero address are considered invalid, and queries
+    ///  about them do throw.
+    /// @return The address of the owner of the deed
+    function ownerOf(uint256 _deedId) external view returns (address _owner);
+
+    /// @notice Transfer ownership of a deed -- THE CALLER IS RESPONSIBLE
+    ///  TO CONFIRM THAT `_to` IS CAPABLE OF RECEIVING DEEDS OR ELSE
+    ///  THEY MAY BE PERMANENTLY LOST
+    /// @dev Throws unless `msg.sender` is the current deed owner, an authorized
+    ///  operator, or the approved address for this deed. Throws if `_from` is
+    ///  not the current owner of the deed. Throws if `_to` is the zero address.
+    ///  Throws if `_deedId` is not a valid deed.
+    /// @param _from The new owner for the deed
+    /// @param _to The new owner for the deed
+    /// @param _deedId The deed to transfer
+    function unsafeTransfer(address _from, address _to, uint256 _deedId) external payable;
+
+    /// @notice Transfers the ownership of a given deed from one address to
+    ///  another address
+    /// @dev Throws unless `msg.sender` is the current deed owner, an authorized
+    ///  operator, or the approved address for this deed. Throws if `_from` is
+    ///  not the current owner of the deed. Throws if `_to` is the zero address.
+    ///  Throws if `_deedId` is not a valid deed. When transfer is complete,
+    ///  this function also calls `onNFTReceived` on `_to` and throws if the return
```

Comment:
For both versions of `transferFrom`, it needs to be made clear that `onNFTReceived` is only called when the code size of `_to` > 0 (i.e `_to` is a contract).

Reply:
Thank you! Updated in e847ffe